metadata-ingestion/tests/unit/test_bigquery_source.py

```diff
@@ -405,3 +406,93 @@ def test_table_processing_logic_date_named_tables(client_mock, data_dictionary_m
     ]  # alternatively
     for table in tables.keys():
         assert tables[table].table_id in ["test-table", "20220103"]
+
+
+@pytest.mark.parametrize(
+    "table_name, expected_table_prefix, expected_shard",
+    [
+        ("project.dataset.table", "project.dataset.table", None),
+        ("table", "table", None),
+        ("project.dataset.table_20231215", "project.dataset.table", "20231215"),
+        ("table_20231215", "table", "20231215"),
+        ("table_1624046611000_name", "table_1624046611000_name", None),
+        ("table_1624046611000", "table_1624046611000", None),
+        # Special case where dataset itself is a sharded table
+        ("20231215", None, "20231215"),
+        #
+        ("project.dataset.table_2023", "project.dataset.table_2023", None),
+        ("table_2023", "table_2023", None),
+        ("table_1624046611", "table_1624046611", None),
+        ("2023", "2023", None),
+    ],
+)
+def test_get_table_and_shard_default(
+    table_name: str, expected_table_prefix: Optional[str], expected_shard: Optional[str]
+) -> None:
+    with patch(
+        "datahub.ingestion.source.bigquery_v2.bigquery_audit.BigqueryTableIdentifier._BIGQUERY_DEFAULT_SHARDED_TABLE_REGEX",
+        "((.+)[_$])?(\\d{8})$",
```

Comment:
can we make the `_BIGQUERY_DEFAULT_SHARDED_TABLE_REGEX` a parameter with a default instead of doing this patch stuff?

Reply:
we can do it but avoiding much refractor here. This will need changes in all calls to this method directly/indirectly.